tests/framework/extensions/workloads/template.go

```diff
@@ -4,18 +4,22 @@ import (
 	"fmt"
 
 	appv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // NewContainer is a contructor that creates a container for a pod template i.e. corev1.PodTemplateSpec
-func NewContainer(containerName, image string, imagePullPolicy corev1.PullPolicy, volumeMounts []corev1.VolumeMount, envFrom []corev1.EnvFromSource) corev1.Container {
+func NewContainer(containerName, image string, imagePullPolicy corev1.PullPolicy, volumeMounts []corev1.VolumeMount, envFrom []corev1.EnvFromSource, command []string, securityContext *corev1.SecurityContext, args []string) corev1.Container {
```

Comment:
this is more of a question for Ugur since he wrote this function; why can't we expect users to directly create their containers from corev1.Contianer? Otherwise I can see another 'createCluster' issue, with tons of parameters as we discover we need more for different test cases.

Reply:
do you have an example for this that you have in mind? IMO it will be easier to use a template instead of directly sending the data every time we make use of containers cc: @caliskanugur